DOC: Revise *dMRIPrep*'s road-map (#147)
nipreps/dmriprep: docs/roadmap.rst

.. include:: links.rst

Development road-map
====================
This road-map serves as a guide for developers as well as a way for us to

> communicate to users and other stake-holders aboout the expectations they should

Reviewer: ```suggestion
communicate to users and other stake-holders about the expectations they should
```

have about the current functionality of the software and future developments.

> If you want to be part of the team developing this road-map, please make sure you

Reviewer: ```suggestion
If you would like to be part of the team developing this road-map, please be sure to
```

> take a look into our `Contributors Guidelines <https://www.nipreps.org/community/CONTRIBUTING>`__.

Reviewer: ```suggestion
read our `Contributors Guidelines <https://www.nipreps.org/community/CONTRIBUTING>`__.
```

> Then, please write one of the developers to be invited to our bi-weekly meetings.

Reviewer: Should we use the dmriprep channel on Mattermost for this instead? It's shown as a badge on the README and might be easier to find than the emails in `.mailmap`.

Author: Yup, that sounds like a good idea.

Author: Do you want to propose an edit?

> Then, please write one of the developers to be invited to our bi-weekly meetings.

Reviewer: ```suggestion
Then, you can contact the developers through the `Mattermost channel <https://mattermost.brainhack.org/brainhack/channels/dmriprep>` to be invited to our bi-weekly meetings.
```

> This road-map proposes a :abbr:`RERO (release early, release often)` planning, scheduling

Reviewer: ```suggestion
This road-map proposes a :abbr:`RERO (release early, release often)` philosophy, scheduling
```

a monthly release until the first stable 1.0 release is reached.

.. important::

    Updated: Dec 18, 2020
    Latest release: `0.3.0 (October 13, 2020) <changes.html#october-13-2020>`__.

Version 0.4 (Before end of 2020)
--------------------------------
Version 0.4 will condense all the outcomes of our sprint towards ISMRM's 2021 abstracts deadline.
This mostly includes house-keeping work, and most prominently, the integration of the *SDCFlows* 2.0
alpha releases, which makes *dMRIPrep* go ahead of *fMRIPrep* in addressing distortions caused by
:math:`B_0` inhomogeneity.

This release will also include Salim's efforts in `#144 <https://github.com/nipreps/dmriprep/pull/144>`__

> to provide with a temporary implementation of head-motion and Eddy-currents correction using

Reviewer: ```suggestion
to provide a temporary implementation of head-motion and Eddy-currents correction using
```

FSL's ``eddy``.

> This temporary solution will be replaced by our 3dSHORE-based algorithm ported from QSIPREP,

Reviewer 2: Can we go ahead and name it yet? I thought Eddy Motion Correction is pretty good...

    - First draft
    - Start development
    - Plan for supporting Derek and Ariel in taking the head-motion correction to the finish line.

Version 0.6 (February, 2021)
----------------------------
#. Head motion correction.

    A SHOREline-based approach, ported from QSIPREP. In cases where the data are
    "shelled", 3dSHORE will be used as the diffusion model. If the data are
    single-shell, we will use SFM as the diffusion model.

#. :math:`B_1` inhomogeneity correction

    - Decide whether it can be brought around from estimation on T1w images
    - Decide whether it should be a default-off option that can be enabled with
      a flag, or else, generate both conversions always.

#. Initiate Phase I of testing

    - Compose our test-bed dataset
    - Document Phase I testing and reporting protocols
    - Start execution

# Continue with the development of ISBI 2021 tutorial

Version 0.7 (March, 2021)
-------------------------
The *noisy month*. This is not a musical event, but a development cycle where we will
focus on the implementation of steps addressing noise in DWI:

#. Identification of outlier measurements (+ imputation?)

> #. Implementation of component-based noise identification techniques

Reviewer: ```suggestion
#. Implementation of component-based noise identification techniques

    - Comparison of multiple approaches including MP-PCA, NLMeans and Patch2Self (`#132 <https://github.com/nipreps/dmriprep/issues/132>`)
```

#. Gibbs-ringing: investigate whether it should be estimated if other techniques
   are in place (i.e., component-based above), and ordering of steps.

> 

Reviewer 3: Should we incorporate work with patch2self here (see #132)?

#. Rician bias modeling.

#. DWI carpet-plot and confounds collation.

#. Testing Phase I execution

#. Final release of the ISBI 2021 tutorial

Version 0.8 (April, 2021)
-------------------------
This release will only address bugfixes conducive to finishing evaluation Phase I,
which should conform a pretty solid ensemble ready for premiere in ISMRM 2021.

Version 0.9 (May, 2021)
-----------------------
#. First official presentation at ISMRM 2021 (should the abstract be accepted)
#. Evaluation Phase II starts.

  - Determine an appropriate dataset
  - Plan for benchmarking experiments (`#121 <https://github.com/nipreps/dmriprep/issues/121>`__)

>   - Start with addressing issues as the are reported

Reviewer: ```suggestion
  - Start with addressing issues as they are reported
```

Version 1.0 (Targetted for September 2021)
------------------------------------------
Wrap-up evaluation Phase II with the first stable release of *dMRIPrep*.

> 

Reviewer 2: I think it might be helpful to clarify that dmriprep is both dataset-adaptive (i.e. to heterogeneous acquisition/encoding schemes, available metadata) and user-configurable (i.e. providing the option to run denoising or not if associated compute cost would be too high, susceptibility distortion correction or not if the available data doesn't support it).

Author: > user-configurable (i.e. providing the option to run denoising or not if associated compute cost would be too high, susceptibility distortion correction or not if the available data doesn't support it).

I would prefer it were not very configurable - and that requires a lot of solid heuristics. But the less human intervention, the better.

Reviewer 2: Fair enough. 

Though I think at least the former (i.e. dataset adaptive) is a key selling point since there are good reasons to think that the same steps are not optimal for all types of dMRI data...
